src/main/java/net/barrage/school/java/ecatalog/web/ProductController.java

```diff
@@ -29,6 +36,34 @@ public List<Product> listProducts() {
         return products;
     }
 
+    @PostMapping
+    public Product createProduct(@Valid @RequestBody Product product) {
+        var createdProduct = productService.saveProduct(product);
```

Comment:
Err.. What merchant this product will belong?

Reply:
I implemented it the way where client will have to put merchant information in request body